### PR TITLE
Fix: Fill styling for sliders

### LINKED ISSUE #177: Missing fill styling in native slider
### Subject of the issue
Since the switch to using the native slider the fill functionality of the slider bar has gone missing.

Screenshot 2023-03-15 at 11 06 53

### Your environment
* Master

### Expected behaviour
Fill functionality returns as per screenshot below

Screenshot 2023-03-15 at 11 06 42

### PR DESCRIPTION
**Pull request by oliverfoster**
fixes #177 

### Fix
* Allow fill styling in slider track

EDIT:
### Fix
* Added index to each model item
* Removed index to value calculations 
* Set up handle sizing
* Corrected fill alignment
* Corrected number alignment
* Corrected slider bar alignment
* Corrected correct answer range alignment
* Nested input fill inside of input track

### CODE REVIEW

**Comment by swashbuck** [comment]
Do we still want to change the color of the fill based on correct vs. incorrect? If so, we can borrow the styles from the deprecated [rangeslider](https://github.com/adaptlearning/adapt-contrib-slider/pull/181/files).



Then, what should we do when the correct answer is a range? Should the fill extend to where the handle is (below 70 here)? Or should the fill span only over the correct range (50 - 90)? Or should we hide the fill entirely here?

**Comment by swashbuck** [comment]
Also wondering if we should drop the _handle_ after the question has been submitted.

**Comment by oliverfoster** [comment]
Red yes.
Should the fill extend to where the handle is (below 70), yes. 
Hide handle, yes.

The fill describes the user's selection (width) and correctness (color). But not the correct range. Otherwise the ui is switching its function.

**Comment by oliverfoster** [comment]
@swashbuck would you mind picking up the remaining work?

**Comment by swashbuck** [comment]
> @swashbuck would you mind picking up the remaining work?

@oliverfoster Sure!

File: less/slider.less [hunk 1]
```diff
@@ -122,11 +130,24 @@
...
   &__widget.show-user-answer.is-incorrect &__number-selection {
     background-color: @validation-error;
     color: @validation-error-inverted;
   }
 
+  &__widget.show-user-answer.is-incorrect &__item-input-fill {
+    background-color: @validation-error;
+  }
+
+  // &__widget.show-correct-answer.is-correct &__item-input-fill,
```
Comment (kirsty-hames) [inline less/slider.less:146]: Is this needed?

File: templates/slider.jsx [hunk 2]
```diff
@@ -173,7 +174,7 @@ export default function Slider (props) {
 
         {/* always present start and end notches */}
         <div className="slider__scale-container js-slider-scale" ref={sliderScaleRef}>
-          <div className="slider__scale-notch slider__scale-notch-start"></div>
+          <div className="slider__scale-notch slider__scale-notch-start" />
           {_showScale &&
               <div className="slider__scale-notch-container js-slider-scale-notch-container">
```
Comment (kirsty-hames) [inline templates/slider.jsx:179]: Very minor and isn't part of this PR but there's an additional indent here.

**Review by kirsty-hames — APPROVED** [review]
Looking good thanks @swashbuck. Tested on Mac Safari, Chrome and FF. iPhone and Android tablet.

**Comment by guywillis** [comment]
Made a few adjustments / optimisations in the plugin and moved the styling parts to the vanilla theme to pick up - https://github.com/adaptlearning/adapt-contrib-vanilla/pull/408

File: less/slider.less (hunk 1 above)
Comment (guywillis) [reply less/slider.less]: Have uncommented out this code and moved it to the vanilla theme

File: templates/slider.jsx (hunk 2 above)
Comment (guywillis) [reply templates/slider.jsx:145]: Reduced indent

**Review by ethan-lp — APPROVED** [review]

**Review by kirsty-hames — APPROVED** [review]
I've just given this another test on the same spec above to review the latest commits. Looks good thanks.

**Comment by oliverfoster** [comment]
> Tested on Windows machine, Chrome, Edge and Firefox, all seems to be working nicely.
> 
> @guywillis @oliverfoster @swashbuck one query, if you answer incorrectly and click show correct answer you see the range of correct answers, if you get the question right you only see your answer, would it make sense to see the range if you answer correctly?

This pr is meant to fix the existing implementation and return it to functionality. Discussions about new behaviour or behaviour changes should be put into a separate issue. But, sure. Sounds like a good idea.

**Review by joe-allen-89 — APPROVED** [review]

**Pull request merged**